src/tests/harness/BenchmarkDotNet.Extensions.Tests/PartitionFilterTests.cs

```diff
@@ -0,0 +1,133 @@
+                artifactsPath: new DirectoryInfo(Path.Combine(Path.GetDirectoryName(typeof(PartitionFilterTests).Assembly.Location), "BenchmarkDotNet.Artifacts")),
+                mandatoryCategories: ImmutableHashSet.Create(Categories.Libraries, Categories.Runtime, Categories.ThirdParty));
+            (bool isSuccess, IConfig parsedConfig, var _) = ConfigParser.Parse(new string[] { "--filter", "*" }, nullLogger, recommendedConfig);
+            Assert.True(isSuccess);
+
+            Assembly microbenchmarksAssembly = typeof(Categories).Assembly;
+            (bool allTypesValid, IReadOnlyList<Type> runnable) = Running.TypeFilter.GetTypesWithRunnableBenchmarks(
+                Array.Empty<Type>(),
+                new Assembly[1] { microbenchmarksAssembly },
+                nullLogger);
+            Assert.True(allTypesValid);
+
+            BenchmarkRunInfo[] allBenchmarks = GetAllBenchmarks(parsedConfig, runnable);
+            Dictionary<string, int> idToPartitionIndex = new ();
+
+            for (int i = 0; i < 10; i++)
+            {
+                Dictionary<string, int> hits = allBenchmarks
+                    .SelectMany(benchmark => benchmark.BenchmarksCases)
+                    .ToDictionary(benchmarkCase => GetId(benchmarkCase), _ => 0);
+
+                for (int partitionIndex = 0; partitionIndex < PartitionCount; partitionIndex++)
+                {
+                    PartitionFilter filter = new(PartitionCount, partitionIndex);
+
+                    foreach (BenchmarkCase benchmark in GetAllBenchmarks(parsedConfig, runnable).SelectMany(benchmark => benchmark.BenchmarksCases))
+                    {
+                        if (filter.Predicate(benchmark))
+                        {
+                            string id = GetId(benchmark);
+
+                            hits[id] += 1;
+
+                            if (idToPartitionIndex.ContainsKey(id))
+                            {
+                                Assert.Equal(partitionIndex, idToPartitionIndex[id]);
+                            }
+                            else
+                            {
+                                idToPartitionIndex.Add(id, partitionIndex);
+                            }
+                        }
+                    }
+                }
+
+                Assert.All(hits.Values, hitCount => Assert.Equal(1, hitCount));
+            }
+
+            static BenchmarkRunInfo[] GetAllBenchmarks(IConfig parsedConfig, IReadOnlyList<Type> runnable)
+            {
+                return runnable
+                    .Select(type => BenchmarkConverter.TypeToBenchmarks(type, parsedConfig))
+                    .Where(benchmark => benchmark.BenchmarksCases.Any())
+                    .ToArray();
+            }
+
+            static string GetId(BenchmarkCase benchmark) => $"{benchmark.Descriptor.Type.Namespace}{benchmark.DisplayInfo}";
+        }
+
+        private class NullLogger : ILogger
```

Comment:
Do you expect this to be used more often? If that's the case, then consider putting it in its own file.

Reply:
@carlossanlop It's very unlikely, as we hardly ever add new BDN extensions (and tests for them).